confidence/io.py

```diff
@@ -234,6 +234,29 @@ def loaders(*specifiers: Locality | Loadable) -> typing.Iterable[Loadable]:
 )
 
 
+def _format_source(source: str | Path, name: str, format: Format) -> Path:
+    match source:
+        case str():
```

Comment:
If there was no difference between the two types of arguments, it would make more sense to apply `str` → `Path` and the follow the same code path from there. Can we make a setup like that work with this deprecation warning in place? 🤔